### PR TITLE
feat: add trex cli for deno

### PR DESCRIPTION
**Pull request by ShawnCockburn**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**

feature

**What is the current behavior? (You can also link to an open issue here)**

trex cli has no autocomplete

**What is the new behavior (if this is a feature change)?**

trex cli has autocomplete

**Additional info:**

Trex is a CLI tool for [Deno](https://deno.land) projects, it is similar to npm, however it follows the deno style of dependencies (inport maps) and adds a npm like script file.

trex website: https://crewdevio.mod.land/projects/Trex

trex repo: https://github.com/crewdevio/Trex

### CODE REVIEW

File: src/trex.ts [hunk 1]
```diff
@@ -0,0 +1,190 @@
+const dependenciesGenerator: Fig.Generator = {
+  script: "cat import_map.json",
+  postProcess: function (out) {
+    if (out && out !== "") {
```
Comment (fedeci) [inline src/trex.ts:4]: This check is enough, in JS an empty string returns false when boolean tested:
```suggestion
    if (out) {
```

File: src/trex.ts [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+      const deps = JSON.parse(out);
```
Comment (fedeci) [inline src/trex.ts:5]: Can we trycatch around this and return an empty array on error?

File: src/trex.ts [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+      }
+    }
+    return [];
+  },
+};
+const scriptsGenerator: Fig.Generator = {
+  script: "cat run.json",
+  postProcess: function (out) {
+    if (out && out !== "") {
+      const scriptsObj = JSON.parse(out);
```
Comment (fedeci) [inline src/trex.ts:26]: Same here

File: src/trex.ts [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+
+        return suggestions;
+      }
+    }
+    return [];
+  },
+};
+
+const trexOptions: Record<string, Fig.Option> = {
+  help: { name: ["-h", "--help"], description: "Print help info" },
```
Comment (fedeci) [inline src/trex.ts:45]: In the spec help does not need to be repeated for every single subcommand but we can use [isPersistent](https://fig.io/docs/reference/option#ispersistent)

File: src/trex.ts [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+      },
+    },
+    {
+      name: "check",
+      description: "Check deno.land [std/x] dependencies updates",
+      options: [
+        trexOptions.help,
+        {
+          name: ["-f", "--fix"],
+          description: "Update outdate dependencies",
```
Comment (fedeci) [inline src/trex.ts:181]:
```suggestion
          description: "Update outdated dependencies",
```

**Review by fedeci — CHANGES_REQUESTED** [review]
Thanks! Some changes and this is ready to be merged.

File: src/trex.ts (hunk 1 above)
Comment (ShawnCockburn) [reply src/trex.ts:4]: Of course, silly me!

File: src/trex.ts (hunk 2 above)
Comment (ShawnCockburn) [reply src/trex.ts:5]: yes

File: src/trex.ts (hunk 4 above)
Comment (ShawnCockburn) [reply src/trex.ts]: This is great to know, thanks!

**Comment by ShawnCockburn** [comment]
@fedeci done!

**Review by fedeci — APPROVED** [review]
Thanks!

**Pull request merged**